cockpit/src/subscriptions-view.jsx

```diff
@@ -98,51 +125,89 @@ class InstalledProducts extends React.Component {
-                        <DescriptionListDescription>{itm.productName}</DescriptionListDescription>
-                    </DescriptionListGroup>
-                    <DescriptionListGroup>
-                        <DescriptionListTerm>{_("Product ID")}</DescriptionListTerm>
-                        <DescriptionListDescription>{itm.productId}</DescriptionListDescription>
-                    </DescriptionListGroup>
-                    <DescriptionListGroup>
-                        <DescriptionListTerm>{_("Version")}</DescriptionListTerm>
-                        <DescriptionListDescription>{itm.version}</DescriptionListDescription>
-                    </DescriptionListGroup>
-                    <DescriptionListGroup>
-                        <DescriptionListTerm>{_("Arch")}</DescriptionListTerm>
-                        <DescriptionListDescription>{itm.arch}</DescriptionListDescription>
-                    </DescriptionListGroup>
-                    <DescriptionListGroup>
-                        <DescriptionListTerm>{_("Status")}</DescriptionListTerm>
-                        <DescriptionListDescription>{status_text}</DescriptionListDescription>
-                    </DescriptionListGroup>
-                    <DescriptionListGroup>
-                        <DescriptionListTerm>{_("Starts")}</DescriptionListTerm>
-                        <DescriptionListDescription>{start_date_text}</DescriptionListDescription>
-                    </DescriptionListGroup>
-                    <DescriptionListGroup>
-                        <DescriptionListTerm>{_("Ends")}</DescriptionListTerm>
-                        <DescriptionListDescription>{end_date_text}</DescriptionListDescription>
-                    </DescriptionListGroup>
-                </DescriptionList>
-            );
+            if (sca_mode) {
+                columns = [
+                    {
+                        title: (<Split>
+                            <SplitItem isFilled>
+                                {itm.productName}
+                            </SplitItem>
+                        </Split>),
+                        header: true,
+                    }
+                ];
+            } else {
+                columns = [
+                    {
+                        title: (<Split>
+                            <SplitItem isFilled>
+                                {itm.productName}
+                            </SplitItem>
+                            <SplitItem>
+                                <Label
+                                    color={subscribed ? "green" : "red"}>{subscribed ? _("Subscribed") : _("Not subscribed")}</Label>
+                            </SplitItem>
+                        </Split>),
+                        header: true,
+                    }
+                ];
+            }
+
+            if (sca_mode) {
+                body = (
+                    <DescriptionList isHorizontal>
+                        <DescriptionListGroup>
```

Comment:
You duplicate a lot of code here. You can rather create a list with the DescriptionListGroups that these two have in common and then append to the array the ones that differ in a condition.

Then you can pass the array directly to the <DescriptionList isHorizontal>{groups}<DescriptionList>

Reply:
You are right. Is it better now?